hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1173,17 +1179,13 @@ int QuectelNcpClient::checkSimCard() {
 int QuectelNcpClient::configureApn(const CellularNetworkConfig& conf) {
     netConf_ = conf;
     if (!netConf_.isValid()) {
-        // First look for network settings based on ICCID
         // FIXME: CCID may fail, need delay here
         HAL_Delay_Milliseconds(1000);
-        auto resp = parser_.sendCommand("AT+CCID");
+        // First look for network settings based on ICCID
         char buf[32] = {};
-        const int ret = CHECK_PARSER(resp.scanf("+CCID: %31s", buf));
-        const int r = CHECK_PARSER(resp.readResult());
-        CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_AT_NOT_OK);
-        if (ret) {
-            netConf_ = networkConfigForIccid(buf, strlen(buf));
-        }
+        auto lenCcid = getIccidImpl(buf, sizeof(buf));
+        CHECK_TRUE(lenCcid > 5, SYSTEM_ERROR_AT_NOT_OK);
```

Comment:
Should this error be called something else? Like `SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED`?

`SYSTEM_ERROR_AT_NOT_OK` is already processed here - https://github.com/particle-iot/device-os/pull/2185/files#diff-ece868c5f58dddb291c6b739de0da632R546

Reply:
I was just trying to keep the error code the same as it was, but I will change to SYSTEM_ERROR_BAD_DATA.